### PR TITLE
CA-356441: Restart sshd to apply sshd configuration

### PR DESCRIPTION
**Pull request by liulinC**
Signed-off-by: Lin Liu <lin.liu@citrix.com>

### CODE REVIEW

**Comment by liulinC** [comment]
Per test result, if we does not restart sshd after configuration update, ssh still works well, 
but if the password is wrong, the error message is promoted only after user logged in.

File: scripts/plugins/extauth-hook-AD.py [hunk 1]
```diff
@@ -248,6 +260,10 @@ def __init__(self, session, args, ad_enabled=True):
         self._update_key_value("GSSAPIAuthentication", value)
         self._update_key_value("GSSAPICleanupCredentials", value)
 
+    def apply(self):
+        super(SshdConfig, self).apply()
+        run_cmd(["/usr/bin/systemctl", "restart", "sshd"])
```
Comment (edwintorok) [inline scripts/plugins/extauth-hook-AD.py:265]: SSHd is smart enough to keep existing connections open over a restart (it uses a child process), but if all you want is to load a new config then `reload` might be better here

Comment (lindig) [reply scripts/plugins/extauth-hook-AD.py:265]: Indeed, `sshd.service` supports `reload-or-restart` which would use `restart` as a fallback if it were not supported.

**Review by lindig — CHANGES_REQUESTED** [review]
Use `systemctl reload-or-restart sshd`

**Comment by liulinC** [comment]
Thanks Edwin and Lindig for the comments, 
Since the feature master is merged to master, I will just close this PR and raise a new PR target to master to followup.

**Pull request closed without merging**